src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/cargo/CargoNet.java

```diff
@@ -250,12 +251,16 @@ public void tick(Block b) {
      * @return The frequency of the given node
      */
     private static int getFrequency(Location node) {
-        try {
-            String str = BlockStorage.getLocationInfo(node).getString("frequency");
-            return str == null ? 0 : Integer.parseInt(str);
-        } catch (Exception x) {
-            SlimefunPlugin.logger().log(Level.SEVERE, x, () -> "An Error occurred while parsing a Cargo Node Frequency (" + node.getWorld().getName() + " - " + node.getBlockX() + "," + node.getBlockY() + "," + +node.getBlockZ() + ")");
+
+        String str = BlockStorage.getLocationInfo(node).getString("frequency");
+        if (str == null) {
+            return 0;
+        }
+        if (!PatternUtils.NUMERIC.matcher(str).matches()) {
```

Comment:
I don't really see the point of putting a regular expression here.

Yes, it validates the data but we can already be certain that it is gonna match 99.9999% of the time since we control what data goes in there.
Only reason why this would not match is if someone manually edited their data files or if a file got corrupted.

So this to my eyes would rather be hurting performance if anything. How on earth does the extra check make it faster though...?

Reply:
Well, we had a try-catch before for the number-format-exception this is why the performance improved.

Reply:
But it should almost never reach the catch block.
Is the existence of the try/catch itself really affecting performance this much??

Reply:
yes, try-catch is expensive.